experanto/datasets.py

```diff
@@ -802,8 +622,19 @@ def __getitem__(self, idx) -> dict:
             chunk_size = self.chunk_sizes[device_name]
             chunk_s = chunk_size / sampling_rate
 
-            times = np.linspace(s, s + chunk_s, chunk_size, endpoint=False)
-            times = times + self.modality_config[device_name].offset
+            # convert everything to int to avoid numerical issues
+            start_time = int(round(s * self.scale_precision))
+            offset = int(
+                round(
+                    sample_dataset.modality_config[device_name].offset
```

Comment:
I am so confused right now, how does this work? 😅 Where is `sample_dataset` defined?
And in general, does it make sense to first check if offset exists for the device config and if not, set offset to 0?